index.rst

```diff
@@ -1,59 +1,367 @@
+   We should resolve this question by the end of the RFC discussion period.
+
+.. note::
+
+   "Curated" calibration datasets that are written from a source-of-truth in an ``obs_*_data`` git repository (rather than generated directly via pipeline processing) are currently written to ``RUN`` collections with names of the form ``<instrument>/calib/curated/<calibDate>``, which are then ingested directly into an ``<instrument>/calib`` ``CALIBRATION`` collection (which clashes with our proposal earlier to make ``<instrument>/calib`` a ``CHAINED`` collection "pointer").
+
+   The full workflow for curated calibrations is sufficiently unclear that it is unlikely that we will get this right in time for the first long-lived Gen3 repository.
+   Initially, our proposal is to use ``RUN`` collections of the form ``<instrument>/calib/<ticket>/curated/<calibDate>``, and a ``CALIBRATION`` collection of the form ``<instrument>/calib/<ticket>`` (which would in general hold non-curated calibrations as well).
+   This leaves room for multiple curated calibration ingests to coexist, which is necessary because they will improve over time, but we don't want to assume we can remove old ones.
+   It does not provide a way to avoid duplication of curated calibration datasets that have not changed.
+
+   Calibration collections created by converting the default Gen2 calibration repo for an instrument will use ``gen2/defaults`` instead of ``<ticket>``, i.e. ``<instrument>/calib/gen2/defaults`` for the ``CALIBRATION`` collection.
+
+
+Filesystem locations
+====================
+
+The main shared data repository for all instruments at NCSA will have a public repository root of ``/datasets/repo``, which will be a symlink to a directory of the form ``/datasets/repo_<YYYYMMDD>``.
+These directories will each contain a ``butler.yaml`` file that points to the appropriate database (with a one-to-one correspondance between databases or database schemas and ``repo_<YYYYMMDD>`` directories).
+
+The default (POSIX) datastore will write datasets with templates that begin with the ``RUN`` name, resulting in e.g. the datasets of per-instrument ``RUN`` collections landing in ``/datasets/repo_<YYYYMMDD>/<instrument>/`` and per-user ``RUN`` collections landing in ``/datasets/repo_<YYYYMMDD>/u/<user>``.
+Users are discouraged from inspecting these directories (as this will be at least quite different in the IDF or other future cloud-based datastores), and *strongly* discouraged from modifying them in any way other than via middleware tools.
+In many cases, write access will actually be prohibited (see :ref:`access-controls`).
+
+When migrations are necessary due to changes in the repository format (something that is *always* preceded by an RFC with explicit CCB approval), a new ``repo_<YYYYMMDD>`` directory and database/schema pair will be created, and files will shared via hard links until/unless the old repository is retired.
```

Comment:
Do folks have something in mind re how to migrate an old database/schema? Does every schema change come with a migration script?

Reply:
Nothing super concrete.  We have started to and are continuing to evaluate [Alembic](https://alembic.sqlalchemy.org/en/latest/) and hope to utilize as much of it as we can.  But our schema is determined from configuration as well as software versions (so _in theory_ it's a big unordered forest of versions, not a nice linear progression or even a tree), and we have a lot of non-schema content that would often need to be migrated as well, and that means we'll either be using a lot of advanced Alembic features or asking it to take care of only a subset of the whole problem.

We do have pretty good mechanisms for making it unnecessary to migrate old repositories immediately, even when code changes would create new repositories with a new schema.  And we've now got pretty good versioning embedded to detect when a migration is needed.  But for the actual migration scripts, I'll defer to @andy-slac, as he's the one who has thought about this the most.  Mostly, we're trying to avoid _any_ migrations for the next few months, to give him time to go do APDB things first and then get back to designing the migration system.